sam-react-app/coreconcept/src/searchform/SearchForm.tsx

```diff
@@ -0,0 +1,43 @@
+import React, {ChangeEvent, Component, KeyboardEvent} from "react";
+
+type SearchFormComponentState = {
+    keyword: string;
+}
+type SearchFormComponentProps = {
+    initialQuery: string;
+    onSearch: (keyword: string)=> void;
+}
+class SearchForm extends Component<SearchFormComponentProps, SearchFormComponentState> {
+    constructor(props: SearchFormComponentProps) {
+        super(props);
+        this.state = {
+            keyword:"",
+        };
+
+        this.updateKeyword = this.updateKeyword.bind(this);
+    }
+    handleOnClick = () => {
+        this.props.onSearch(this.state.keyword);
+    }
+    handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+        if(event.key === "Enter"){
+            this.props.onSearch(this.state.keyword);
+        }
+    }
+    updateKeyword(event: ChangeEvent<HTMLInputElement> ){
+        this.setState({
+            keyword: event.target.value
+        })
+    }
+    render(){
+        return <div>
+            <p>-------------- Search form component --------------</p>
+            <input type="textbox" defaultValue={this.props.initialQuery} onChange={this.updateKeyword}
```

Comment:
You created an [uncontrollable component (see docs)](https://legacy.reactjs.org/docs/uncontrolled-components.html) here. Basically what it means is that `input` doesn't receive its value from the state. `defaultValue` is not a react attribute but a standard HTML one and defines, as one might expect, default value for that input. Now, when you type something in this input you call `onChange` and it updates state, but changes are not reflected back to `input`. In this situation if something else updates `keyword` in state `input` will remain the same. In order to keep inputs (any inputs: text, select, radio and so on) in sync with the state you need to add `value` attribute. You can check [this page](https://legacy.reactjs.org/docs/forms.html) for more details.

Reply:
done